firefox-ios/Client/Frontend/Home/UnifiedAds/UnifiedAdsProvider.swift

```diff
@@ -134,6 +142,25 @@ final class UnifiedAdsProvider: URLCaching, UnifiedAdsProviderInterface, Feature
         }
     }
 
+    private func fetchTilesWithAdsClient(completion: @escaping (UnifiedTileResult) -> Void) {
+        self.logger.log("Fetching tiles with ads client", level: .info, category: .homepage)
+        let mozAdRequests = [
+            MozAdsPlacementRequest(placementId: TileOrder.position1.rawValue, iabContent: nil),
+            MozAdsPlacementRequest(placementId: TileOrder.position2.rawValue, iabContent: nil)
+        ]
+        do {
+            let mozAdsImages = try self.adsClient.requestImageAds(mozAdRequests: mozAdRequests, options: nil)
+            let unifiedTiles: [UnifiedTile] = mozAdsImages.map { name, mozAdsImage in
+                UnifiedTile.from(name: name, mozAdsImage: mozAdsImage)
+            }
+            self.logger.log("Ads client request successful", level: .info, category: .homepage)
+            completion(.success(unifiedTiles))
+        } catch let error {
+            self.logger.log("Ads client request failed: \(error)", level: .warning, category: .homepage)
+            completion(.failure(Error.noDataAvailable))
```

Comment:
nit; it's not needed to use `self.` in this case. We prefer to keep `self.` usage where it's needed like in closures
```suggestion
        logger.log("Fetching tiles with ads client", level: .info, category: .homepage)
        let mozAdRequests = [
            MozAdsPlacementRequest(placementId: TileOrder.position1.rawValue, iabContent: nil),
            MozAdsPlacementRequest(placementId: TileOrder.position2.rawValue, iabContent: nil)
        ]
        do {
            let mozAdsImages = try self.adsClient.requestImageAds(mozAdRequests: mozAdRequests, options: nil)
            let unifiedTiles: [UnifiedTile] = mozAdsImages.map { name, mozAdsImage in
                UnifiedTile.from(name: name, mozAdsImage: mozAdsImage)
            }
            logger.log("Ads client request successful", level: .info, category: .homepage)
            completion(.success(unifiedTiles))
        } catch let error {
            logger.log("Ads client request failed: \(error)", level: .warning, category: .homepage)
            completion(.failure(Error.noDataAvailable))
```